src/libfreenect2.cpp

```diff
@@ -736,6 +743,57 @@ void Freenect2DeviceImpl::setIrAndDepthFrameListener(libfreenect2::FrameListener
     pipeline_->getDepthPacketProcessor()->setFrameListener(ir_frame_listener);
 }
 
+void Freenect2DeviceImpl::setColorAutoExposure(float exposure_compensation)
+{
+  CommandTransaction::Result result;
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_ACS, 0u), result);
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_EXPOSURE_MODE, 0u), result);  // 0 == Fully auto
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_EXPOSURE_COMPENSATION, exposure_compensation), result);
+}
+
+void Freenect2DeviceImpl::setColorSemiAutoExposure(float pseudo_exposure_time_ms)
+{
+  CommandTransaction::Result result;
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_ACS, 0u), result);
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_EXPOSURE_MODE, 3u), result);  // 3 == Semi-auto
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_EXPOSURE_TIME_MS, pseudo_exposure_time_ms), result);
+}
+
+void Freenect2DeviceImpl::setColorManualExposure(float integration_time_ms, float analog_gain)
+{
+  CommandTransaction::Result result;
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_ACS, 0u), result);
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_EXPOSURE_MODE, 4u), result);  // 4 == Fully manual
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_INTEGRATION_TIME, integration_time_ms), result);
+  command_tx_.execute(ColorSettingCommand(ColorSettingCommand::SET_ANALOG_GAIN, analog_gain), result);
+}
+
+void Freenect2DeviceImpl::setColorSetting(uint32_t cmd, uint32_t value)
+{
+  CommandTransaction::Result result;
+  command_tx_.execute(ColorSettingCommand((ColorSettingCommand::COMMAND_TYPE)cmd, value), result);
+}
+
+void Freenect2DeviceImpl::setColorSetting(uint32_t cmd, float value)
+{
+  CommandTransaction::Result result;
+  command_tx_.execute(ColorSettingCommand((ColorSettingCommand::COMMAND_TYPE)cmd, value), result);
+}
+
+uint32_t Freenect2DeviceImpl::getColorSetting(uint32_t cmd)
+{
+  CommandTransaction::Result result;
+  command_tx_.execute(ColorSettingCommand((ColorSettingCommand::COMMAND_TYPE)cmd), result);
+  ColorSettingResponse const* data = reinterpret_cast<ColorSettingResponse const*>(&result[0]);
```

Comment:
Must check if result.size() > sizeof(ColorSettingResponse) before doing the cast here.